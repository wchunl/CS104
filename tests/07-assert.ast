ROOT "" (0.0.0)
|   TOK_FUNCTION "" (6.22.1) function
|   |   TOK_TYPE_ID "" (6.22.1) typeid
|   |   |   TOK_VOID "void" (6.22.1) void
|   |   |   TOK_IDENT "fail" (6.22.6) variable
|   |   TOK_PARAM "(" (6.22.11) param
|   |   |   TOK_TYPE_ID "" (6.22.12) typeid
|   |   |   |   TOK_STRING "string" (6.22.12) string
|   |   |   |   TOK_IDENT "expr" (6.22.19) string variable (6.22.12)
|   |   |   TOK_TYPE_ID "" (6.22.25) typeid
|   |   |   |   TOK_STRING "string" (6.22.25) string
|   |   |   |   TOK_IDENT "file" (6.22.32) string variable (6.22.25)
|   |   |   TOK_TYPE_ID "" (6.22.38) typeid
|   |   |   |   TOK_INT "int" (6.22.38) int
|   |   |   |   TOK_IDENT "line" (6.22.42) int variable (6.22.38)
|   TOK_FUNCTION "" (6.24.1) function
|   |   TOK_TYPE_ID "" (6.24.1) typeid
|   |   |   TOK_VOID "void" (6.24.1) void
|   |   |   TOK_IDENT "putchr" (6.24.6) variable
|   |   TOK_PARAM "(" (6.24.13) param
|   |   |   TOK_TYPE_ID "" (6.24.14) typeid
|   |   |   |   TOK_INT "int" (6.24.14) int
|   |   |   |   TOK_IDENT "chr" (6.24.18) int variable (6.24.14)
|   TOK_FUNCTION "" (6.25.1) function
|   |   TOK_TYPE_ID "" (6.25.1) typeid
|   |   |   TOK_VOID "void" (6.25.1) void
|   |   |   TOK_IDENT "putint" (6.25.6) variable
|   |   TOK_PARAM "(" (6.25.13) param
|   |   |   TOK_TYPE_ID "" (6.25.14) typeid
|   |   |   |   TOK_INT "int" (6.25.14) int
|   |   |   |   TOK_IDENT "num" (6.25.18) int variable (6.25.14)
|   TOK_FUNCTION "" (6.26.1) function
|   |   TOK_TYPE_ID "" (6.26.1) typeid
|   |   |   TOK_VOID "void" (6.26.1) void
|   |   |   TOK_IDENT "putstr" (6.26.6) variable
|   |   TOK_PARAM "(" (6.26.13) param
|   |   |   TOK_TYPE_ID "" (6.26.14) typeid
|   |   |   |   TOK_STRING "string" (6.26.14) string
|   |   |   |   TOK_IDENT "str" (6.26.21) string variable (6.26.14)
|   TOK_FUNCTION "" (6.28.1) function
|   |   TOK_TYPE_ID "" (6.28.1) typeid
|   |   |   TOK_INT "int" (6.28.1) int
|   |   |   TOK_IDENT "getchr" (6.28.5) variable
|   |   TOK_PARAM "(" (6.28.11) param
|   TOK_FUNCTION "" (6.29.1) function
|   |   TOK_TYPE_ID "" (6.29.1) typeid
|   |   |   TOK_STRING "string" (6.29.1) string
|   |   |   TOK_IDENT "getstr" (6.29.8) variable
|   |   TOK_PARAM "(" (6.29.14) param
|   TOK_FUNCTION "" (6.30.1) function
|   |   TOK_TYPE_ID "" (6.30.1) typeid
|   |   |   TOK_STRING "string" (6.30.1) string
|   |   |   TOK_IDENT "getln" (6.30.8) variable
|   |   TOK_PARAM "(" (6.30.13) param
|   TOK_FUNCTION "" (7.5.1) function
|   |   TOK_TYPE_ID "" (7.5.1) typeid
|   |   |   TOK_INT "int" (7.5.1) int
|   |   |   TOK_IDENT "main" (7.5.5) variable
|   |   TOK_PARAM "(" (7.5.9) param
|   |   TOK_BLOCK "{" (7.5.12)
|   |   |   TOK_BLOCK "{" (7.6.4)
|   |   |   |   TOK_IF "if" (7.6.5)
|   |   |   |   |   TOK_NOT "not" (7.6.9)
|   |   |   |   |   |   TOK_EQ "==" (7.6.24) vreg
|   |   |   |   |   |   |   TOK_STRINGCON ""nullptr"" (7.6.14) string const
|   |   |   |   |   |   |   TOK_NULLPTR "nullptr" (7.6.27) null const
|   |   |   |   |   TOK_CALL "(" (7.6.42)
|   |   |   |   |   |   TOK_IDENT "fail" (7.6.37) variable
|   |   |   |   |   |   TOK_STRINGCON ""\"nullptr\" == nullptr"" (7.6.43) string const
|   |   |   |   |   |   |   TOK_STRINGCON ""07-assert.oc"" (7.6.69) string const
|   |   |   |   |   |   |   TOK_INTCON "6" (7.6.85) int const
|   |   |   ';' ";" (7.6.89)
|   |   |   TOK_RETURN "return" (7.7.4)
|   |   |   |   TOK_INTCON "1" (7.7.11) int const
