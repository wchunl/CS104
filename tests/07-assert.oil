fail:     .function
          .param ptr expr
          .param ptr file
          .param int line
          .end
putchr:   .function
          .param int chr
          .end
putint:   .function
          .param int num
          .end
putstr:   .function
          .param ptr str
          .end
getchr:   .function int
          .end
getstr:   .function
          .end
getln:    .function
          .end
main:     .function int
.if0:     $t0:i = "nullptr" == nullptr
          $t1:i = not $t0:i
          goto .el0 if not $t1:i
.th0:     $t2:i = "\"nullptr\" == nullptr"
          fail($t2:i)
.fi0:     return 1
          return 
          .end
