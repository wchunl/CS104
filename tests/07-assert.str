string_set[   0]:   12808232827189627082 0x1526ef8->"nullptr"
                    4706206694200543916 0x1524dc8->","
                   16579916199511856337 0x1526488->"getln"
string_set[   3]:   11868120863286193964 0x15266f8->"main"
                    9611428142580219387 0x1520758->"fail"
string_set[   4]:   12654391512392035424 0x1526d98->""nullptr""
                      21606623943142546 0x1526998->"{"
                    4017852018407011111 0x1524ed8->"file"
string_set[   6]:   13686721134187162920 0x1527278->"}"
string_set[   8]:    9386170080689500517 0x1524c78->"expr"
string_set[   9]:    5475249891658879256 0x15270e8->""\"nullptr\" == nullptr""
string_set[  12]:   15816506932648033945 0x1524ae8->"("
string_set[  14]:    9317282556992397945 0x1525268->"if"
string_set[  17]:   18413295556186398326 0x15252c8->"putint"
string_set[  18]:    7211130878751977441 0x1525418->"putchr"
                   12679825168387489523 0x15259a8->"num"
string_set[  19]:    3835993668518112351 0x1527298->"6"
string_set[  23]:   11166701219533395393 0x15261f8->"getstr"
                   12868445110721718657 0x1524ef8->")"
string_set[  28]:   14635384904416252381 0x15273c8->"return"
string_set[  29]:   12291392672316406109 0x15250b8->"line"
string_set[  31]:   14736352883882276958 0x1526e48->"=="
                    6142509188972423790 0x1520738->""
                   13348923631108115385 0x1525738->"putstr"
string_set[  32]:   16459639625557179573 0x15271c8->""07-assert.oc""
string_set[  33]:    1583576014053361350 0x1525b58->"str"
string_set[  35]:   14849783028996024063 0x1526c88->"not"
string_set[  38]:   12597418120160561823 0x1524cf8->"int"
string_set[  39]:    7296690267450279874 0x1525588->"chr"
string_set[  40]:   11911796782241914524 0x1520318->"void"
string_set[  42]:    5344662657899890615 0x15250d8->";"
string_set[  44]:   13497578857207247412 0x1524b98->"string"
string_set[  45]:   10159970873491820195 0x1527468->"1"
string_set[  46]:    9201800968376702500 0x1525f88->"getchr"
load_factor = 0.723
bucket_count = 47
max_bucket_size = 3
