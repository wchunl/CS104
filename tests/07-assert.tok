#  1 "07-assert.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "07-assert.oc"
#  1 "oclib.h"
# 22 "oclib.h"
   6 22.00001 261 TOK_VOID      void
   6 22.00006 282 TOK_IDENT     fail
   6 22.00011  40 '('           (
   6 22.00012 263 TOK_STRING    string
   6 22.00019 282 TOK_IDENT     expr
   6 22.00023  44 ','           ,
   6 22.00025 263 TOK_STRING    string
   6 22.00032 282 TOK_IDENT     file
   6 22.00036  44 ','           ,
   6 22.00038 262 TOK_INT       int
   6 22.00042 282 TOK_IDENT     line
   6 22.00046  41 ')'           )
   6 22.00047  59 ';'           ;
   6 24.00001 261 TOK_VOID      void
   6 24.00006 282 TOK_IDENT     putchr
   6 24.00013  40 '('           (
   6 24.00014 262 TOK_INT       int
   6 24.00018 282 TOK_IDENT     chr
   6 24.00021  41 ')'           )
   6 24.00022  59 ';'           ;
   6 25.00001 261 TOK_VOID      void
   6 25.00006 282 TOK_IDENT     putint
   6 25.00013  40 '('           (
   6 25.00014 262 TOK_INT       int
   6 25.00018 282 TOK_IDENT     num
   6 25.00021  41 ')'           )
   6 25.00022  59 ';'           ;
   6 26.00001 261 TOK_VOID      void
   6 26.00006 282 TOK_IDENT     putstr
   6 26.00013  40 '('           (
   6 26.00014 263 TOK_STRING    string
   6 26.00021 282 TOK_IDENT     str
   6 26.00024  41 ')'           )
   6 26.00025  59 ';'           ;
   6 28.00001 262 TOK_INT       int
   6 28.00005 282 TOK_IDENT     getchr
   6 28.00011  40 '('           (
   6 28.00012  41 ')'           )
   6 28.00013  59 ';'           ;
   6 29.00001 263 TOK_STRING    string
   6 29.00008 282 TOK_IDENT     getstr
   6 29.00014  40 '('           (
   6 29.00015  41 ')'           )
   6 29.00016  59 ';'           ;
   6 30.00001 263 TOK_STRING    string
   6 30.00008 282 TOK_IDENT     getln
   6 30.00013  40 '('           (
   6 30.00014  41 ')'           )
   6 30.00015  59 ';'           ;
#  4 "07-assert.oc"
   7 5.00001 262 TOK_INT       int
   7 5.00005 282 TOK_IDENT     main
   7 5.00009  40 '('           (
   7 5.00010  41 ')'           )
   7 5.00012 123 '{'           {
   7 6.00004 123 '{'           {
   7 6.00005 265 TOK_IF        if
   7 6.00008  40 '('           (
   7 6.00009 281 TOK_NOT       not
   7 6.00013  40 '('           (
   7 6.00014 285 TOK_STRINGCON "nullptr"
   7 6.00024 275 TOK_EQ        ==
   7 6.00027 270 TOK_NULLPTR   nullptr
   7 6.00034  41 ')'           )
   7 6.00035  41 ')'           )
   7 6.00037 282 TOK_IDENT     fail
   7 6.00042  40 '('           (
   7 6.00043 285 TOK_STRINGCON "\"nullptr\" == nullptr"
   7 6.00067  44 ','           ,
   7 6.00069 285 TOK_STRINGCON "07-assert.oc"
   7 6.00083  44 ','           ,
   7 6.00085 283 TOK_INTCON    6
   7 6.00086  41 ')'           )
   7 6.00087  59 ';'           ;
   7 6.00088 125 '}'           }
   7 6.00089  59 ';'           ;
   7 7.00004 268 TOK_RETURN    return
   7 7.00011 283 TOK_INTCON    1
   7 7.00012  59 ';'           ;
   7 8.00001 125 '}'           }
