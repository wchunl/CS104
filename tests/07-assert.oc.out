; 0.  1: # 1 "07-assert.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "07-assert.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: # 1 "oclib.h" 1
; 5.  1: # 22 "oclib.h"
; 6. 22: void fail (string expr, string file, int line);
; 6. 23: 
; 6. 24: void putchr (int chr);
; 6. 25: void putint (int num);
; 6. 26: void putstr (string str);
; 6. 27: 
; 6. 28: int getchr();
; 6. 29: string getstr();
; 6. 30: string getln();
; 6. 31: # 4 "07-assert.oc" 2
; 7.  4: 
; 7.  5: int main() {
; 7.  6:    {if (not ("nullptr" == nullptr)) fail ("\"nullptr\" == nullptr", "07-assert.oc", 6);};
; 7.  7:    return 1;
; 7.  8: }
